lib/monitor.toit

```diff
@@ -169,17 +170,20 @@ monitor Signal:
 
   // Helper method for condition waiting.
   wait_ [condition] -> none:
-    while true:
-      awaited := awaited_
-      if current_ == awaited:
```

Comment:
This is not enough to guarantee that nobody is waiting. The first `wait` after a signal will always hit this even if there are waiters where the condition hasn't yet been met.